torch_geometric/loader/neighbor_loader.py

```diff
@@ -50,11 +51,15 @@ def __init__(
                     f"'time_attr' attribute not yet supported for "
                     f"'{data.__class__.__name__}' object")
 
-            # Convert the graph data into a suitable format for sampling.
-            out = to_csc(data, device='cpu', share_memory=share_memory,
-                         is_sorted=is_sorted)
-            self.colptr, self.row, self.perm = out
-            assert isinstance(num_neighbors, (list, tuple))
+            if not has_edges(data):
+                self.num_neighbors = []
+                self.colptr = self.row = self.perm = None
```

Comment:
Why do we need the branching here? If the graph has no edges, than all these dictionaries will just be empty, correct?

Reply:
`to_csc` doesn't work if `data` doesn't have edges. Also `colptr` and `row` aren't dictionaris.

Reply:
Ah yes, I think we could just create `edge_index = torch.empty((2, 0), dtype=torch.long)` in `to_csc` in case it does not exist?

Reply:
Yep made that fix, it works. Thanks @rusty1s.